client/apps/edit/components/content2/index.coffee

```diff
@@ -10,14 +10,15 @@ module.exports = React.createClass
 
   componentWillMount: ->
     @debouncedSave = _.debounce((->
-      @props.article.save()
+      @props.article.save() # TODO: Why is this not saving updates from #19?
```

Comment:
@eessex - this thing is almost totally complete but for some reason whenever line 19 fires (for example when uploading an image) it eventually calls `@debouncedSave` but doesn't save the latest values. Logging out in 19 below everything is coming through right, but for some reason when I log here there is no `cover_image` on the model.